source/containers.tex

```diff
@@ -3414,7 +3414,8 @@
 \indextext{requirements!container}%
 An \tcode{array} meets all of the requirements of a container and
 of a reversible container\iref{container.requirements}, except that a default
-constructed \tcode{array} object is not empty and that \tcode{swap} does not have constant
+constructed \tcode{array} object is not empty, and
+that move construction and \tcode{swap} do not have constant
```

Comment:
The requirements table for general container requirements already says (in Note A and Note B) that the complexity in the array case is linear, not constant. So, it feels we should make the "swap" exception as well as the move construction exception a note here.

Reply:
Maybe after #4460, which doesn't use notes anymore. From reading https://www.iso.org/sites/directives/current/part2/index.xhtml#_idTextAnchor333, it seems that these meant to not be normative. But they're referred to by normative text right after the table. Maybe we need a better term to not reuse note.

In any case, it doesn't feel right to make the exception in the requirements themselves rather than the specification of the container. To say that `array` _does_ meet the requirements, make no exception after saying so (or sneaking that in a note), to then make exceptions in the requirements feels the wrong way to go about it.

If we were to go with your suggestion here, how about including the whole sentence to also include the default constructor, which doesn't have a note in the requirements table (in the WD nor in #4460)? I suppose that then we'd need a Note C for the default constructor in the requirements table similar to the Note A and B.

Reply:
Those "Note A/B" things are certainly normative, despite the spelling of "note".

Reply:
Done. I suppose the next step would be to move the remaining exception to the referred requirements for consistency.

Reply:
Yes, except that's unfriendly to do while #4460 is in flight.